qf-faucet-server/src/main.rs

```diff
@@ -0,0 +1,288 @@
+        })?;
+
+    let account_array: [u8; 32] = account_bytes[..32].try_into()
+        .map_err(|e| {
+            println!("Error: {:?}", e);
+            warp::reject::custom(SomeError)
+        })?;
+    let rpc_endpoint = config.rpc_url;
+    let api = OnlineClient::<PolkadotConfig>::from_url(rpc_endpoint)
+        .await
+        .map_err(|e| {
+            println!("Error: {:?}", e);
+            warp::reject::custom(SomeError)
+        })?;
+
+    // Create a keypair from the provided address
+    let mnemonic = env::var("MNEMONIC").expect("MNEMONIC is not set");
+    let phrase = Mnemonic::from_str(mnemonic.as_str()).map_err(|e| {
+        println!("Error: {:?}", e);
+        warp::reject::custom(SomeError)
+    })?;
+
+    let from;
+    if config.debug {
+        println!("Transfer from Alice");
+        from = dev::alice();
+    } else {
+        from = sr25519::Keypair::from_phrase(&phrase, None).map_err(|e| {
+            println!("Error: {:?}", e);
+            warp::reject::custom(SomeError)
+        })?;
+    }
+    
+    let dest = MultiAddress::Id(AccountId32(account_array));
+
+
+    let transfer = polkadot::tx().balances().transfer_keep_alive(dest, FAUCET_AMOUNT.into());
+
+    let events = api
+        .tx()
+        .sign_and_submit_then_watch_default(&transfer, &from)
+        .await
+        .map_err(|e| {
+            println!("Error Submit transfer: {:?}", e);
+            warp::reject::custom(TransferError)
+        })?
+        .wait_for_finalized_success()
+        .await
+        .map_err(|e| {
+            println!("Error transfer not finalized: {:?}", e);
+            warp::reject::custom(TransferError)
+        })?;
+    
+    let _ = store_transfer(body.address.as_str(), 2, format!("{}", events.extrinsic_hash()).as_str())
+        .map_err(|e| {
+            println!("Error store the transfer: {:?}", e);
+            warp::reject::custom(StorageError)
+        })?;
+
+    Ok(warp::reply::json(&format!("Extrinsic submitted: {:?}", events.extrinsic_hash())))
+}
+
+#[derive(Parser, Debug)]
+#[command(name = "QF faucet bot server")]
+#[command(version = "1.0")]
+#[command(about = "QF server for crediting by dev tokens", long_about = None)]
+struct Cli {
+    /// Host can set from env HOST or this option [default: 0.0.0.0]
+    #[arg(long = "host", short = 'H')]
+    ip: Option<String>,
+    /// Port can set from env PORT or this option [default: 8080]
+    #[arg(short = 'P', long)]
+    port: Option<String>,
+    /// RPC url can set from env RPC_URL or this option [default: wss://dev.qfnetwork.xyz/socket]
+    #[arg(short, long)]
+    rpc_url: Option<String>,
+    /// In debug mode the sender is Alice
+    #[arg(short, long, action)]
+    debug: bool,
+    /// Custom delay for transfer in minutes
+    #[arg(short, long, default_value = "120")]
+    timeout: u64,
+}
+
+#[derive(Clone)]
+struct ServerConfig {
+    rpc_url: String,
+    debug: bool,
+    timeout: Option<u64>,
```

Comment:
It's never `None` since we have `default_value = "120"`:

```suggestion
    timeout: u64,
```

Reply:
Not fixed

Reply:
Resolved